src/commands/classified/external.rs

```diff
@@ -184,88 +222,109 @@ async fn run_with_stdin(
     input: Option<InputStream>,
     is_last: bool,
 ) -> Result<Option<InputStream>, ShellError> {
-    let name_tag = command.name_tag;
     let home_dir = dirs::home_dir();
+    let path = context.shell_manager.path()?;
+    let name_tag = command.name_tag.clone();
+    let args = command.args.clone();
 
-    let mut process = Exec::cmd(&command.name);
+    let inputs = input.unwrap_or_else(InputStream::empty).into_vec().await;
```

Comment:
If we had a pipeline like `cat really_large_file | my_external`, we could very easily OOM. Any reason we need to `into_vec()` here instead of `await`-ing one item at a time in the stream below? As a user of a shell, I want my shell to be incredibly resilient to these kinds of things (it's okay if `my_external` were to OOM though).

Reply:
+1 for awaiting and emitting

Reply:
Oops. We had this statement for the `$it` case that I removed. I left this one deliberately here. What should we do for stdin? Should we await each row like we do in `$it` case and feed each row as stdin to the external independently? or should we await each row and fold it to one single value to write as stdin to the external (current behavior)?

Reply:
Yeah that's a good idea

Reply:
c

Reply:
👍 for one row at a time. I think the one thing we should strive for in `nushell` is avoiding panics/failures. I think users of shells expect that out of their shell 🙂